### PR TITLE
DAOS-12183 docker: Run DAOS Agent service in a container

### PR DESCRIPTION
**Pull request by knard-intel**
# Description

Defines how to use docker containers for running an application accessing a DAOS file system.
The DAOS Agent service and the DAOS client are run on two containers over the same docker engine.

Using DocOnly Pragma as the docker things are not tested by the CI.

# Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

# Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by knard-intel** [comment]
@ashleypittman , @mark-olson and @wiliamhuang could you tell me if you will have the time to review this PR.
No problem, if you are too overloaded at this time, I will look for other reviewers.
Thanks in advance.

**Review by mark-olson — COMMENTED** [review]
**Name of the `utils/docker/cloud` directory**

The content in this directory:

- does not contain a utility
- is not related to anything in cloud

  This directory contains examples for using containers.

  Maybe the `utils/docker/cloud` directory could be named `utils/docker/examples` to better relflect the content within.

**Comment by knard-intel** [comment]
> **Name of the `utils/docker/cloud` directory**
> 
> The content in this directory:
> 
> * does not contain a utility
> * is not related to anything in cloud
>   This directory contains examples for using containers.
>   Maybe the `utils/docker/cloud` directory could be named `utils/docker/examples` to better relflect the content within.

- [ ] Change the name of the directiory

File: utils/docker/cloud/.env [hunk 1]
```diff
@@ -8,10 +8,21 @@
 # DAOS_CLIENT_UID="666"
 # DAOS_CLIENT_GNAME="bar"
 # DAOS_CLIENT_GID="999"
+# DAOS_ACCESS_POINTS="['hostname1', 'hostname2']"
 
 # XXX Arguments which must be updated if needed XXX
 DAOS_AGENT_RUNTIME_DIR="/var/run/daos_agent"
```
[the remark this answers is not in the document]
Comment (knard-intel) [reply utils/docker/cloud/.env:14]: - [ ] Move `DAOS_AGENT_RUNTIME_DIR` to optional section

**Review by mark-olson — APPROVED** [review]

**Review by wiliamhuang — APPROVED** [review]

**Pull request merged**